pontoon/base/views.py

```diff
@@ -655,6 +660,10 @@ def update_translation(request):
                 t.fuzzy = False
 
             t.save()
+            t.warnings.clear()
+            t.errors.clear()
+
+            save_failed_checks(t, failed_checks)
```

Comment:
This is confusing me. In `save_failed_checks()` we [delete](https://github.com/mozilla/pontoon/pull/963/files#diff-8c894a1d32099aa55fbfe944f8d1ac65R32) translation warnings and errors, why would we need to disassociate them first here?

Also, this call to `save_failed_checks()` is the only one where we might have existig warnings and errors (because translation exists), so it makes more sense to move the logic here.